src/PayID/pay-id-client.ts

```diff
@@ -25,16 +31,10 @@ export default class PayIDClient implements PayIDClientInterface {
    * @returns An XRP address representing the given PayID.
    */
   async xrpAddressForPayID(payID: string): Promise<string> {
-    const paymentPointer = PayIDUtils.parsePaymentPointer(payID)
-    if (!paymentPointer) {
-      throw PayIDError.invalidPaymentPointer
-    }
-
-    // Swagger generates the '/' in the URL by default and the payment pointer's 'path' is prefixed by a '/'. Strip off the leading '/'.
-    const path = paymentPointer.path.substring(1)
+    const payIDComponents = PayIDClient.parsePayID(payID)
```

Comment:
Refactored this logic to a private static helper method to share across methods.